onnxruntime/core/providers/cuda/nn/batch_norm.cc

```diff
@@ -21,10 +21,19 @@ namespace cuda {
       (*KernelDefBuilder::Create())                               \
           .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
       BatchNorm<T>);                                              \
+  ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_EX(                       \
+      BatchNormalization,                                         \
+      kOnnxDomain,                                                \
+      9, 13,                                                      \
+      T,                                                          \
+      kCudaExecutionProvider,                                     \
+      (*KernelDefBuilder::Create())                               \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
+      BatchNorm<T>);                                              \
   ONNX_OPERATOR_TYPED_KERNEL_EX(                                  \
       BatchNormalization,                                         \
       kOnnxDomain,                                                \
-      9,                                                          \
+      14,                                                         \
```

Comment:
It looks like this kernel registration has a small bug. Opset-14 introduced a new type constraint 'U' for mean/var. The implementation only supports cases where T == U. For T != U cases, it will crash within the kernel when we reference data of type T for the inputs containing type U. I guess we need to add `.TypeConstraint("U",DataTypeImpl::GetTensorType<T>())` here.